### PR TITLE
Fix for WFMP-138, Integrate galleon provisioning

### PR DESCRIPTION
**Pull request by jfdenise**
* Introduced new Maven goals
* Replaced download of artifacts
* Removed dependency on wildfly-cli and jboss-modules. Done by reflection based on CLI artifact resolved at runtime. NB: This introduce some complexity in the plugin src code. I am wandering if we could, in 3.0, fork CLI execution in its own process in all cases and not support in-process CLI execution?
* Added resolve-expressions (false by default) when executing CLI commands (PackageMojo and ExecuteCommandsMojo).

### CODE REVIEW

File: plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java [hunk 1]
```diff
@@ -72,81 +77,46 @@
...
-     *
-     * @deprecated this will be removed in 3.0.0 in favor of provisioning a server
+     * The version of the WildFly default server to install in case no jboss-home has been set
+     * and no server has previously been provisioned.
+     * The latest stable version is resolved if left blank.
      */
-    @Deprecated
-    @Parameter(property = PropertyNames.WILDFLY_CLASSIFIER)
-    private String classifier;
+    @Parameter(alias = "version", property = PropertyNames.WILDFLY_VERSION)
```
Comment (jamezp) [inline plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java:111]: Not a huge deal, but `alias = "version"` shouldn't really be needed.

File: plugin/src/main/java/org/wildfly/plugin/common/StandardOutput.java [hunk 2]
```diff
@@ -100,22 +104,46 @@ private StandardOutput(final Target target, final OutputStream consumerStream, f
...
+    public static StandardOutput parse(final String stdout, final boolean discardNone, final boolean append) throws IOException {
         if (stdout == null) {
             return new StandardOutput(Target.INHERIT, null, Redirect.INHERIT, null);
         }
         final Target target;
         Path stdoutPath = null;
         final OutputStream out;
         final String value = stdout.trim().toLowerCase(Locale.ENGLISH);
-        if ("system.out".equals(value)) {
+        if (SYSTEM_OUT.equals(value)) {
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply plugin/src/main/java/org/wildfly/plugin/common/StandardOutput.java]: Yep, fixed.

File: plugin/src/main/java/org/wildfly/plugin/provision/PackageServerMojo.java [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+     * &lt;properties-files&gt;<br/>
+     * &lt;property-file&gt;my-properties.properties&lt;/property-file&gt;<br/>
+     * &lt;/properties-files&gt;<br/>
+     * &lt;java-opts&gt;<br/>
+     * &lt;java-opt&gt;-Xmx256m&lt;/java-opt&gt;<br/>
+     * &lt;/java-opts&gt;<br/>
+     * &lt;!-- Expressions resolved during server execution --&gt;<br/>
+     * &lt;resolve-expressions&gt;false&lt;/resolve-expressions&gt;<br/>
+     * &lt;/packaging-script&gt;<br/>
+     * &lt;/packaging-scripts&gt;
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply plugin/src/main/java/org/wildfly/plugin/provision/PackageServerMojo.java]: Thanks fixed.

File: plugin/src/main/java/org/wildfly/plugin/provision/PackageServerMojo.java [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+        IoUtils.recursiveDelete(log);
+    }
+
+    private MavenModelControllerClientConfiguration getClientConfiguration() {
+        return null;
+    }
+
+    private ModelControllerClient createClient() {
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply plugin/src/main/java/org/wildfly/plugin/provision/PackageServerMojo.java]: Removed methods.

File: plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java [hunk 5]
```diff
@@ -72,81 +77,46 @@
  */
 @Mojo(name = "run", requiresDependencyResolution = ResolutionScope.RUNTIME)
 @Execute(phase = LifecyclePhase.PACKAGE)
-@SuppressWarnings("DeprecatedIsStillUsed")
 public class RunMojo extends AbstractServerConnection {
 
-    @Parameter(defaultValue = "${project}", readonly = true, required = true)
-    private MavenProject project;
+@Component
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java]: Thanks, fixed.

File: plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java (hunk 1 above)
Comment (jfdenise) [reply plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java]: Thanks, fixed.

File: pom.xml [hunk 6]
```diff
@@ -77,13 +89,12 @@
...
     <properties>
         <!-- WildFly/JBoss dependencies -->
-        <version.org.jboss.jdf.stacks-client>1.0.3.Final</version.org.jboss.jdf.stacks-client>
         <version.org.jboss.logging.jboss-logging>3.4.2.Final</version.org.jboss.logging.jboss-logging>
         <version.org.jboss.logmanager>2.1.10.Final</version.org.jboss.logmanager>
         <version.org.wildfly.checkstyle-config>1.0.8.Final</version.org.wildfly.checkstyle-config>
         <version.org.wildfly.common.wildfly-common>1.5.4.Final</version.org.wildfly.common.wildfly-common>
+        <!-- This version property is also retrieved by plugin at runtime to resolve CLI artifact -->
         <version.org.wildfly.core>17.0.1.Final</version.org.wildfly.core>
-        <version.org.wildfly.dist>24.0.1.Final</version.org.wildfly.dist>
```
[the remark this answers is not in the document]
Comment (jfdenise) [reply pom.xml:86]: Yes, I introducd it back, it controls version of the provisioned servers for tests.

**Pull request merged**